GeoQuiz/app/src/main/java/com/example/geoquiz/QuizActivity.kt

```diff
@@ -52,43 +72,113 @@ class QuizActivity : AppCompatActivity() {
             /*val toast=Toast.makeText(this, "Correct",Toast.LENGTH_SHORT)
             toast.setGravity(Gravity.TOP,0,20)
             toast.show()*/
+
             checkAnswer(true)
+
         }
 
         mFalseButton.setOnClickListener {
+
             checkAnswer(false)
+
         }
 
+
         mNextButton.setOnClickListener {
             mCurrentIndex=(mCurrentIndex+1)%mQuestionBank.size
             /*val question=mQuestionBank[mCurrentIndex].textResId
             mQuestionTextView.setText(question)*/
+
             updateQuestion()
+
+
         }
 
         mPrevButton.setOnClickListener {
             mCurrentIndex=(mCurrentIndex-1)%mQuestionBank.size
+            if(mCurrentIndex<0){
+                mCurrentIndex=mQuestionBank.size-1
+            }
             updateQuestion()
+
         }
 
         updateQuestion()
+
+        scoreButton.setOnClickListener {
+
+            for( i in correctAnswers){
+                score += 1;
+            }
+
+
+            Toast.makeText(this,"Your score is: ${(score.toDouble()/4)*100}%",Toast.LENGTH_SHORT).show()
+
+        }
+
+
```

Comment:
For the end client, code formatting and styling does not really matter as long as the app works. For us developers however, it helps our eyes if code is arranged and cleanly structured. You have a lot of newlines in here. Please consider auto-formatting this with the help of Cmd + Option + L